packages/firestore/src/api/database.ts

```diff
@@ -406,6 +410,20 @@ export class Firestore implements firestore.FirebaseFirestore, FirebaseService {
     );
   }
 
+  clearPersistence(): Promise<void> {
+    if (this.isClientRunning) {
+      throw new FirestoreError(
+        Code.FAILED_PRECONDITION,
+        'Persistence cannot be cleared while the client is running'
+      );
+    }
+    const persistenceKey = IndexedDbPersistence.buildStoragePrefix(
+      this.getDatabaseInfo()
+    );
+    const dbName = persistenceKey + IndexedDbPersistence.MAIN_DATABASE;
+    return IndexedDbPersistence.clearPersistence(dbName);
```

Comment:
I would suggest that you generate the `dbName` in IndexedDbPersistence and only pass the persistence key to `clearPersistence()`.